File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/core/LastDoKitViewPosInfo.java
package com.didichuxing.doraemonkit.kit.core;

import com.didichuxing.doraemonkit.util.ScreenUtils;

/**
 * ================================================
 * 作    者：jint（金台）
 * 版    本：1.0
 * 创建日期：2019-10-29-10:25
 * 描    述：保存上一次DokitView的位置信息
 * 修订历史：
 * ================================================
 */
public class LastDoKitViewPosInfo {
    private boolean isPortrait = true;
    private int doKitViewWidth;
    private int doKitViewHeight;
    private float leftMarginPercent;
    private float topMarginPercent;

    public int getDoKitViewWidth() {
        return doKitViewWidth;
    }

    public void setDoKitViewWidth(int doKitViewWidth) {
        this.doKitViewWidth = doKitViewWidth;
    }

    public int getDoKitViewHeight() {
        return doKitViewHeight;
    }

    public void setDoKitViewHeight(int doKitViewHeight) {
        this.doKitViewHeight = doKitViewHeight;
    }

    public void setPortrait() {
        this.isPortrait = ScreenUtils.isPortrait();
    }

    public void setLeftMargin(int leftMargin) {
        this.leftMarginPercent = (float) leftMargin / (float) ScreenUtils.getAppScreenWidth();
    }

    public void setTopMargin(int topMargin) {
        this.topMarginPercent = (float) topMargin / (float) ScreenUtils.getAppScreenHeight();
    }

    public boolean isPortrait() {
        return isPortrait;
    }

    public float getLeftMarginPercent() {
        return leftMarginPercent;
    }

    public float getTopMarginPercent() {
        return topMarginPercent;
    }

    @Override
    public String toString() {
        return "LastDokitViewPosInfo{" +
                "isPortrait=" + isPortrait +
                ", leftMarginPercent=" + leftMarginPercent +
                ", topMarginPercent=" + topMarginPercent +
                '}';
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/core/LastDokitViewPosInfo.kt
package com.didichuxing.doraemonkit.kit.core

import com.blankj.utilcode.util.ScreenUtils

/**
 * ================================================
 * 作    者：jint（金台）
 * 版    本：1.0
 * 创建日期：2019-10-29-10:25
 * 描    述：保存上一次DokitView的位置信息
 * 修订历史：
 * ================================================
 */
class LastDokitViewPosInfo {
    val isPortrait = ScreenUtils.isPortrait()
    var dokitViewWidth = 0
    var dokitViewHeight = 0
    var leftMarginPercent = 0f
        private set
    var topMarginPercent = 0f
        private set


    fun setLeftMargin(leftMargin: Int) {
        leftMarginPercent = leftMargin.toFloat() / ScreenUtils.getAppScreenWidth().toFloat()
    }

    fun setTopMargin(topMargin: Int) {
        topMarginPercent = topMargin.toFloat() / ScreenUtils.getAppScreenHeight().toFloat()
    }

    override fun toString(): String {
        return "LastDokitViewPosInfo{" +
                "isPortrait=" + isPortrait +
                ", leftMarginPercent=" + leftMarginPercent +
                ", topMarginPercent=" + topMarginPercent +
                '}'
    }
}